### ISSUE TITLE
memory leak using find

### ISSUE THREAD

**Issue by YinianFan** [Automattic/mongoose#11283]
**Do you want to request a *feature* or report a *bug*?**

bug

**What is the current behavior?**

When finding thousands of documents, each has some keys and an array of thousands of ObjectId of other documents. The memory usage grows when finding those documents and never released after finding.

**If the current behavior is a bug, please provide the steps to reproduce.**

schema in schedule.js
```
import mongoose from 'mongoose'
const Schema = mongoose.Schema
const scheduleSchema = new Schema({
  _id: { type: Schema.Types.String, default: constructObjectId },  // constructObjectId transform ObjectId to string
  eventId: { type: Schema.Types.String }, 
  topic: { type: Schema.Types.String, require: true }, 
  action: { type: Schema.Types.String }, 
  system: { type: Schema.Types.String, require: true }, 
  status: { type: Schema.Types.String, default: "" }, 
  category: { type: Schema.Types.Number, default: 0 }, 
  peopleId: { type: Schema.Types.String }, 
  tenantId: { type: Schema.Types.String }, 
  content: { 
    title: { type: Schema.Types.String }, 
    content: { type: Schema.Types.String }, 
    remark: { type: Schema.Types.String }, 
  },
  url: { type: Schema.Types.String }, 
  operations: [{ 
    title: { type: Schema.Types.String }, 
    action: { type: Schema.Types.Number, default: 0 }, 
    url: { type: Schema.Types.String, default: "" }, 
  }],
  startDateTime: { type: Date }, 
  endDateTime: { type: Date }, 
  isSingleDay: { type: Schema.Types.Boolean }, 
  duration: { type: Schema.Types.Number }, 
  conflictList: [{ type: Schema.Types.String, ref: 'schedule' }],  // has thousands of other documents' ObjectId
  optimisticLock: { type: Date }, 
  deleted: { type: Schema.Types.Boolean, default: false }
}, {
  versionKey: false,
  timestamps: { createdAt: 'createTime', updatedAt: 'updateTime' }
})
const schedule = mongoose.model('schedule', scheduleSchema)
module.exports = schedule
```
find: 
```
import schedule from './schedule.js'
exports.find = async (query) => {
  // do something
  let filter = {
    "startDateTime": { $lt: endMoment.toDate() },
    "endDateTime": { $gt: startMoment.toDate() },
    "peopleId": peopleId,
    "deleted": false,
    "tenantId": { $in: activateTenant },
    "isSingleDay": true
  }
  let schedules= await schedule.find(filter)
 // do something with schedules and return result
}
```
Grafana moniter shows the memory usage grows when finding
memory usage grow from 150MB to 2GB, and never released
![image](https://user-images.githubusercontent.com/16176690/151288280-66f3cb30-8321-40ad-9fd5-d56aa41726aa.png)

**What is the expected behavior?**

no memory leak

**What are the versions of Node.js, Mongoose and MongoDB you are using? Note that "latest" is not a version.**
Node.js 12.18.0
Mongoose 5.12.12
MongoDB 4.2.8

**Comment by lorand-horvath** [comment]
Update Mongoose to 5.13.14 - there have been many fixes since 5.12.12
https://github.com/Automattic/mongoose/blob/5.x/History.md

**Comment by YinianFan** [comment]
Upgrade Mongoose to 5.13.14 and problem solved, thanks a lot
